src/darwin/Framework/CHIP/MTROperationalCredentialsDelegate.mm

```diff
@@ -228,19 +226,15 @@
 {
     if (operationalCertificate == nil || intermediateCertificate == nil || rootCertificate == nil) {
         setNSError(CHIP_ERROR_INVALID_ARGUMENT, error);
-        return;
     }
 
     // use ipk and adminSubject from CommissioningParameters if not passed in.
     // Dereferencing mCppCommissioner as it would be set to point to a valid Cpp commissioner by now, as we are in the middle of
     // commissioning
-    __block chip::Optional<chip::Controller::CommissioningParameters> commissioningParameters;
-    dispatch_sync(mChipWorkQueue, ^{
-        commissioningParameters = mCppCommissioner->GetCommissioningParameters();
-    });
+    chip::Optional<chip::Controller::CommissioningParameters> commissioningParameters
+        = mCppCommissioner->GetCommissioningParameters();
```

Comment:
This does not look right.  We're not on the Matter queue here, are we?

Reply:
Dispatched sync on chipWorkQueue.